playbooks/roles/airship-deploy-tempest/tasks/main.yml

```diff
@@ -0,0 +1,116 @@
+---
+# tasks file for airship-deploy-tempest
+- name: Load standard variables
+  include_vars: "{{ playbook_dir }}/../vars/common-vars.yml"
+  tags:
+    - always
+
+- name: Set site path
+  set_fact:
+    site_path: "{{ upstream_repos_clone_folder }}/airship/treasuremap/site/{{ socok8s_site_name }}"
+  tags:
+    - always
+
+- name: Ensure site directory for {{ socok8s_site_name }} exists
+  become: yes
+  file:
+    path: "{{ site_path }}"
+    state: directory
+
+- name: Ensure tempest chart directory exists
+  become: yes
+  file:
+    path: "{{ site_path }}/software/charts/osh/openstack-tempest"
+    state: directory
+
+- name: Determine tempest test type (smoke or all)
+  set_fact:
+    test_args: "{% if tempest_test_type == 'all' %}--regex 'tempest.api.*'{% else %}--smoke{% endif %}"
```

Comment:
What about scenario tests, shouldn't they be part of `all`?

In my option `test_args` could be placed at `defaults/main.yml` like this:

```
tempest_test_args:
  smoke:  "--smoke"
  all: "--regex .*"
```

and on `tempest.yaml`:
```
...
script: "tempest run --config-file /etc/tempest/tempest.conf -w 4 {{ tempest_test_args[tempest_test_type] }}"
...
```

Reply:
Excellent idea, I've made that change.